includes/reader-activation/class-reader-data.php

```diff
@@ -348,17 +373,14 @@ public static function check_newsletter_subscription( $timestamp, $data ) {
 		if ( ! empty( $is_newsletter_subscriber ) && gettype( $is_newsletter_subscriber ) === 'string' ) {
 			$is_newsletter_subscriber = json_decode( $is_newsletter_subscriber );
 		}
-		// Bail if reader is already a newsletter subscriber.
-		if ( $is_newsletter_subscriber ) {
```

Comment:
If `$is_newsletter_subscriber` is not used anymore, lines above defining & mutating it can also be removed.